ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Types.hs

```diff
@@ -111,47 +111,47 @@ newtype ChainDbHandle m blk = CDBHandle (StrictTVar m (ChainDbState m blk))
 
 -- | Check if the ChainDB is open, if so, executing the given function on the
 -- 'ChainDbEnv', otherwise, throw a 'CloseDBError'.
-getEnv :: forall m blk r. (IOLike m, HasCallStack)
+getEnv :: forall m blk r. (IOLike m, HasCallStack, StandardHash blk, Typeable blk)
```

Comment:
In this file I would also go for the shorter:
```suggestion
getEnv :: forall m blk r. (IOLike m, HasCallStack, HasHeader blk)
```
Many other functions in this file mention the same constraints.